### PR TITLE
[PM-13170] [PM-13171] Enable use of new generators on web

### PR DESCRIPTION
**Pull request by djsmith85**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-13170

## 📔 Objective

Create a web-specific hosting component for the new credential-generator. ~~Temporarily, this is going to use the new password-generator component, but will be replaced once https://github.com/bitwarden/clients/pull/11398 has merged.~~

The swap is guarded by a feature flag, so this can be merged without affecting current production.

### Before
image

### After
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/tools/credential-generator/credential-generator.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<app-header></app-header>
+
+<bit-container>
+  <!-- Will get replaced with <tools-credential-generator /> once https://github.com/bitwarden/clients/pull/11398 has been merged -->
```
Comment (audreyality) [inline apps/web/src/app/tools/credential-generator/credential-generator.component.html:4]: ⛏️ I've been calling out things to address in later commits using `FIXME: xyz`. That way it's easy to find tech debt with `grep`.

**Review by audreyality — COMMENTED** [review]
👍🏻  Lookin' good! Just a nit.

File: apps/web/src/app/tools/credential-generator/credential-generator.component.html (hunk 1 above)
Comment (djsmith85) [reply apps/web/src/app/tools/credential-generator/credential-generator.component.html]: Addressed with ff18d01

**Review by audreyality — APPROVED** [review]

**Review by audreyality — APPROVED** [review]

File: apps/web/src/app/shared/loose-components.module.ts [hunk 2]
```diff
@@ -221,6 +223,7 @@ import { SharedModule } from "./shared.module";
     OrgUserConfirmComponent,
     OrgWeakPasswordsReportComponent,
     GeneratorComponent,
+    CredentialGeneratorComponent,
```
Comment (coroiu) [inline apps/web/src/app/shared/loose-components.module.ts:226]: issue: we should avoid adding new components to this module

**Review by coroiu — CHANGES_REQUESTED** [review]

File: apps/web/src/app/shared/loose-components.module.ts (hunk 2 above)
Comment (djsmith85) [reply apps/web/src/app/shared/loose-components.module.ts]: @coroiu You are absolutely right and as it's a standalone component, I didn't need to import here anyway 🙈 

Addressed with 605d017

**Review by coroiu — APPROVED** [review]
Platform approval no longer needed - approving to get rid of "requested changes status"

File: apps/web/src/app/shared/loose-components.module.ts (hunk 2 above)
Comment (coroiu) [reply apps/web/src/app/shared/loose-components.module.ts]: Hehe it was the only platform dependency too :D Approved!

**Review by audreyality — APPROVED** [review]

**Pull request merged**